lib/net.js

```diff
@@ -954,7 +954,10 @@ function afterConnect(status, handle, req, readable, writable) {
     return;
   }
 
-  assert(handle === self._handle, 'handle != self._handle');
+  // Update handle if it was wrapped
+  handle = self._handle;
+  // TODO(indutny): assert that the handle is actually an ancestor of old one
+  // assert(handle === self._handle, 'handle != self._handle');
```

Comment:
I think the `Readble`/`Writable` should be swapped with this direction of arrows, but generally this is correct.